File: backend/app/schemas/message.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MessageOut(BaseModel):
    id: int
    user_id: int
    sender_type: str
    message: Optional[str]
    message_type: str
    voice_url: Optional[str]
    voice_duration: Optional[int]
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class SendMessageRequest(BaseModel):
    message: str

File: frontend/types/message.ts
export interface Message {
  id: number
  user_id: number
  sender_type: 'user' | 'admin'
  message: string | null
  message_type: 'text' | 'voice'
  voice_url: string | null
  voice_duration: number | null
  is_read: boolean
  created_at: string
}
